pkg/config/types.go

```diff
@@ -2,110 +2,17 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
```

Comment:
Should we move the remaining types into `provider.go`?

Reply:
probably yep, I split them first by `interface`/`struct`, but it makes little sense, they are both kind of interface